src/llvm_ts.cpp

```diff
@@ -287,10 +286,31 @@ static void submit_cpu(KernelType type, Func &&func, uint32_t width,
     jitc_task = new_task;
 }
 
-Task *LLVMThreadState::launch(Kernel kernel, uint32_t size,
-                              std::vector<void *> *kernel_params,
-                              uint32_t /* kernel_param_count */,
-                              const uint8_t* /* kernel_params_global */) {
+/// Temporary scratch space for scheduled tasks
+static std::vector<Task *> scheduled_tasks;
+
+void LLVMThreadState::barrier() {
+    if (scheduled_tasks.size() == 1) {
+        task_release(jitc_task);
```

Comment:
I don't know about the LLVM backends and the task system, but is this safe to do at any point?
If not, the docstring of `ThreadState::barrier()` should specify exactly when it's safe & intended to call this function.

Reply:
Good point.
To me it seems fine to call it at any point, but I'm also not 100% sure.

Reply:
I guess it just makes sense to call it after a set of kernel launches.  All the other operations have implicit barriers between them.